src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject.cs

```diff
@@ -64,7 +65,16 @@ public override void Disconnect()
         protected void AddFile(string filename, SourceCodeKind sourceCodeKind)
         {
             bool getIsCurrentContext(IVisualStudioHostDocument document) => LinkedFileUtilities.IsCurrentContextHierarchy(document, RunningDocumentTable);
-            AddFile(filename, sourceCodeKind, getIsCurrentContext, GetFolderNamesFromHierarchy);
+            var itemid = Hierarchy?.TryGetItemId(filename) ?? VSConstants.VSITEMID_NIL;
```

Comment:
if TryGetItemId checks whether Hierarchy is null and return NIL, this can be just Hierarchy.TryGet...

Reply:
I personally hate extension methods that accept null receivers. Yeah, prior to us having ? that was convenient, but this is the future. We have ? now.

Reply:
I would also prefer we not have them.  Except for maybe '.IsXXX' methods, where it's clear they're meant to be like ```is```, and thus can work on anything.  Also, because if you use ```?.``` here you get a ```bool?``` which is pretty annoying to work with vs a standard bool.

Reply:
> I would also prefer we not have them

Didn't you write most of them in our codebase? :smile:

Dunno, in this case I don't mind the bool?-ness at all. The default fallback this code is assuming (Nil) is plainly stated and understood. It's not the shortest code, but means you don't have to look at three files to see what's going on. That's always how I prefer to optimize.

Reply:
> Didn't you write most of them in our codebase? 

Yes :)   Prior to having ?.    I'm ok with not having them now that we do.  I just was pointing out that i think it's pretty painful for the ```isErs```.  :)